### ISSUE TITLE
It is hard to add extra functionality to sinks that already exists

### ISSUE THREAD

**Issue by xyz1001** [gabime/spdlog#2318]
spdlog version: 1.x
I want to add extra functionality to `spdlog::sinks::daily_file_sink_mt`. But I found it is hard.
First, I cannot inherit `spdlog::sinks::daily_file_sink_mt` because it is a `final` class.
Then I try to create a class `MySink` which inherit `spdlog::sinks::sink` and pass a instance of `spdlog::sinks::daily_file_sink_mt` to `MySink`, just like decorator pattern. The problem is that `spdlog::sinks::sink` has non virtual member functions, so I have no way to invoke same function in `spdlog::sinks::daily_file_sink_mt` when one call these functions on `MySink`.
Is there any better way to implement it? Any reply is appreciated.

**Comment by gabime** [comment]
I would just copy paste the daily file sink, make the needed adjustments and call it mysink.

**Comment by xyz1001** [comment]
I will accept your advice. However, it is hard to follow changes from repo.
